### PR TITLE
Perf stats v1

### PR DESCRIPTION
**Pull request by brando90**
Hi @gwicke ,

I finished implementing and debugging the add k largest things from the database. I tested it, and as far as I know it should be working (i.e. should be keeping track of the k largest and stuff)

Note that I have specified in the settings file the number of top values we are keeping track of but I believe that the settings file is being ignored by git, so I am not sure if u can see that change.

Also, I just realized that we are adding them to the Cassandra db but we also need to get these results at some point. To make it clear, I implemented keeping track of the top k and have implemented a temporary version to actually query these top k results that I would like feedback on.

Currently I implemented something but not sure if we might want to change the API. Comment on this so that I know what to do.

Currently, there are 7 types of results data that can be obtained from a single processed result:

For time:

-total
-wt2html
-html2wt

For size:
-htmlraw
-htmlgzip
-wtraw
-wtgzip

All of these have their top k values stored on the DB, but the db needs to know which one you want to query. How should the server interact with the CassandraBackend to specify which one it want? 

1) One way could be that we only specify the commit and it returns all 7 for the specific commit. 
2) The other way is to specify both the name of the type of result we want and the commit (current implementation)

But, what i was thinking is adding more code to the add function because, it would be kind of annoying to have to specify in the schema in advance, what those tables should look like. One way to make this more flexible is, to write some logic that does the following:

If you try to add a result, but that result does not have a table, create a database with cassandra and then you can do normal operation. I am more in favour with that implementation since it would make it more fexible for you guys since you can add more results value or drop them as you wish and the logic can take care of resolving those additions without human intervention.

What do you think?

Also, I just realized, for API (1) to work, we would also need to keep another table of the names of the types of times we keep track of, so that it could query them all automatically without needing human intervention. Or we could only allow API (2) and we dont have to worry about that. But the implementations are not disjoint, both of them could be implemented or (2) could be a subroutine/helper function for (1)

### CODE REVIEW

File: insert_function.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  var middle = array[index];
+  if( element <= middle ){
+    return positionToInsertRec(array, element, i, index - 1);
+  }else{
+    return positionToInsertRec(array, element, index + 1, j);
+  }
+}
+
+/**
+* recursive function that finds the position to which to insert an element.
```
Comment (brando90) [inline insert_function.js]: Wrong comment, I need to fix this. Or can be fixed during merge. Remove the word recursive.

**Comment by brando90** [comment]
@gwicke  updated the comment, I realized it wasn't all that clear what I wrote so I re wrote one part just in case you need/want clarification on any part I wrote.

File: CassandraBackend.js [hunk 2]
```diff
@@ -138,12 +146,17 @@ function initTestPQ(commitIndex, numTestsLeft, cb) {
     if (!lastHash) {
       cb(null);
     }
-    var cql = 'select test, score, commit from test_by_score where commit = ?';
+    var cql = 'select test, score, commit from test_by_score where commit = ?'; //TODO this doesnt have a ; at the end of cql. issue?
```
Comment (gwicke) [inline CassandraBackend.js:222]: It's better to terminate cql queries, but afaik the library also does so implicitly.

**Comment by brando90** [comment]
@gwicke is there still a problem with this pull? I have already merged it before and I have fetch from upstream and git says there is nothing to merge. Are you having merging problems from your side? Because I am not.

**Pull request closed without merging**